lib/consts/consts.go

```diff
@@ -10,22 +10,45 @@ import (
 // Version contains the current semantic version of k6.
 const Version = "0.46.0"
 
-// VersionDetails can be set externally as part of the build process
-var VersionDetails = "" //nolint:gochecknoglobals
-
 // FullVersion returns the maximally full version and build information for
 // the currently running k6 executable.
 func FullVersion() string {
 	goVersionArch := fmt.Sprintf("%s, %s/%s", runtime.Version(), runtime.GOOS, runtime.GOARCH)
-	if VersionDetails != "" {
-		return fmt.Sprintf("%s (%s, %s)", Version, VersionDetails, goVersionArch)
+
+	buildInfo, ok := debug.ReadBuildInfo()
+	if !ok {
+		return fmt.Sprintf("%s (%s)", Version, goVersionArch)
+	}
+
+	var (
+		commit string
+		dirty  bool
+	)
+	for _, s := range buildInfo.Settings {
+		switch s.Key {
+		case "vcs.revision":
```

Comment:
We even have `vcs.time` so ... I guess we can even add that if somebody wants :shrug: